lib/compilers/cerberus.ts

```diff
@@ -33,6 +38,54 @@ import {CompilationEnvironment} from '../compilation-env.js';
 import {logger} from '../logger.js';
 import * as utils from '../utils.js';
 
+type Range = {
+    start: Parser.Point;
+    end: Parser.Point;
+};
+
+type LocationRange = {
+    location: Range;
+    cursor: Range | Parser.Point | null;
+};
+
+type ASTwithLocations = {
+    root: Parser.SyntaxNode;
+    locations: Map<number, LocationRange>;
+};
+
+type SyntaxNodeWithLocation = {
+    node: Parser.SyntaxNode;
+    loc: LocationRange | undefined;
+};
+
+/*
```

Comment:
Can you remove this commented-out code please? Or if it's needed, comment it back in and add tests to exercise it.